File: learning dsa/LinearSearch.java
import java.util.*;
public class LinearSearch {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0;i<n;i++){
            arr[i] = sc.nextInt();
        }
        System.out.println("Enter the element you want to search : ");
        int x = sc.nextInt();
        for(int i=0;i<n;i++){
            if(arr[i]==x){
                System.out.println("Element found at index "+ i);
                break;
            }
            else{
                System.out.println("Element not found");
                break;
            }
        }

    }
    
}

File: linearsearch.cpp
#include<iostream>
using namespace std;
int main()
{
    int n;
    cout<<"\nEnter the size of an array";
    cin>>n;
    int a[n];
    for(int i=0;i<n;i++){
        cout<<"\nEnter the element at index "<<i<<" is :";
        cin>>a[i];
    }
    cout<<"\narray are :";
    for(int i=0;i<n;i++){
        cout<<a[i]<<" ";

    }
    int k,index=-1;
    cout<<"\nEnter the value you want to search";
    cin>>k;
    for(int i=0;i<n;i++){
        if(k==a[i]){
            index=i;
            cout<<"element found at index "<<i;
        }
    
        if(index==-1){
            cout<<"Element not found";
        }
    }
}